src/ngx_http_lua_rewriteby.c

```diff
@@ -164,24 +164,62 @@ ngx_http_lua_rewrite_handler(ngx_http_request_t *r)
 
 
 ngx_int_t
-ngx_http_lua_rewrite_handler_inline(ngx_http_request_t *r)
+ngx_http_lua_rewrite_handler_sets(ngx_http_request_t *r)
 {
-    lua_State                   *L;
     ngx_int_t                    rc;
     ngx_http_lua_loc_conf_t     *llcf;
+    ngx_array_t                 *p_sets;    /*   rewrite sets, inline
+                                                script/script file path */
+    ngx_uint_t                   i;
+    ngx_http_lua_rewrites_t     *rewrite;
 
-    dd("rewrite by lua inline");
+    dd("rewrite by lua handler sets");
+
+    rc = NGX_DECLINED;
 
     llcf = ngx_http_get_module_loc_conf(r, ngx_http_lua_module);
+
+    p_sets = llcf->rewrites;
+
+    for (i = 0; i < p_sets->nelts; ++i)
+    {
+        rewrite = (ngx_http_lua_rewrites_t *)((char *)p_sets->elts
```

Comment:
We usually do not index an element of `ngx_array_t` object in the nginx world. Please use the common idiom like below:

```
elem_type_t   *foo;
foo = array->elts;
for (i = 0; i < array->nelts; i++) {
    /* then index the element like this: */
    foo[i]
}
```